.github/workflows/docs.yml

```diff
@@ -0,0 +1,35 @@
+name: Docs verification
+
+# Trigger the workflow on pull requests and direct pushes to any branch
+on:
+  push:
+  pull_request:
+
+jobs:
+  testdata:
+    name: Verify Docs update
+    runs-on: ubuntu-latest
+    # Pull requests from the same repository won't trigger this checks as they were already triggered by the push
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    steps:
+      - name: Clone the code
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.19"
+      - name: Remove pre-installed kustomize
+        # This step is needed as the following one tries to remove
+        # kustomize for each test but has no permission to do so
+        run: sudo rm -f /usr/local/bin/kustomize
+      - name: Verify docs
+        run: |
+          make generate-docs
```

Comment:
See that CI has reported [an error for missing `kubebuilder`](https://github.com/kubernetes-sigs/kubebuilder/actions/runs/4081288605/jobs/7034545030#step:5:34). That is possibly due to the execution of `make generate-docs` without `build_kb`.

I think we might have two approaches:
1. In the new ci yaml file, add the setup to load and run `build_kb`, and then execute `make generate-docs`.
2. Drop this new ci file, and add `generate-docs` to `make generate`.

WDYT? @Eileen-Yu @camilamacedo86 

Well, I like the idea to have a separate pipeline focusing on docs validation.

Reply:
I think we should not use `build_kb` and we should do == make generate and check testdata

Reply:
Hi @camilamacedo86
It makes sense to integrate the docs generation into `check-the-testdata`, that can make the implementation a lot easier and straightforward.
Based on that, I'm concerned about some following points:
1. If CI fails at `check the testdata`, will it become hard to tell at the first glance which part is wrong: either `testdata` OR `docs`? 
2. Would it be slow if we run all the generations in one pipeline versus having multiple pipelines for each generation?
3. Is the priority of keeping `testdata` consistent higher than `docs`?

Reply:
It is OK to be another CI test. I am just saying that the implementation is very similar == to the target to generate the testdata samples and check it out.

Reply:
Hi @camilamacedo86 I just added CI for the docs check.
Could you pls help with review?

@Kavinjsir It sounds good to have multiple pipelines. How would you like to deal with this in some future PR?